site/frontend/src/pages/compare/compile/table/benchmark-detail.vue

```diff
@@ -48,41 +48,47 @@ const DAY_RANGE = 30;
  * Calculates the start and end range for a history graph for this benchmark
  * and artifact.
  */
-function getGraphRange(artifact: ArtifactDescription): GraphRange {
-  const date = new Date(artifact.date);
-
+function getGraphRange(
+  artifact: ArtifactDescription,
+  baseArtifact?: ArtifactDescription
+): GraphRange {
   // If this is a try commit, we don't know its future, so always we just display
   // the last `DAY_RANGE` days.
   if (artifact.type === "try") {
+    const date = new Date(artifact.date);
     return {
       start: getPastDate(date, DAY_RANGE),
       end: artifact.commit,
       date: null,
     };
   } else {
+    let [start_date, end_date] = [baseArtifact, artifact].map(
+      (c) => c?.date && new Date(c.date)
+    );
     // If this is a master commit, then we try to display `dayRange` days
     // "centered" around the commit date.
 
     // Calculate the end of the range, which is commit date + half of the
     // amount of days we want to show. If this date is in the future,
     // the server will clip the result at the current date.
-    const end = getFutureDate(date, DAY_RANGE / 2);
+    const end = getFutureDate(end_date, DAY_RANGE / 2);
 
     // Calculate how many days there have been from the commit date
     const daysInFuture = Math.min(
       DAY_RANGE / 2,
-      daysBetweenDates(date, new Date())
+      daysBetweenDates(end_date, new Date())
     );
 
     // Calculate how many days we should go into the past, taking into account
     // the days that will be clipped by the server.
-    const daysInPast = DAY_RANGE - daysInFuture;
+    const daysInPast = start_date ? DAY_RANGE / 2 : DAY_RANGE - daysInFuture;
```

Comment:
This is a bit too complicated, and I think that we don't want to go further 15 days back if the start date is e.g. 2 years ago :) I would change the logic so that it chooses `earlier_of(start_date, commit_date - DAY_RANGE / 2)`.

Reply:
Oh, I just read too quickly and misunderstood the comment in the issue 😅.  So, to clarify, here, the `commit_date` is the date of the artifact, namely the `end_date`, right?

Reply:
Yes :)

Reply:
Then I think it's more reasonable that we don't format the date to a date string in both `getPastDate` and `getFutureDate` and export `format_ymd` as `formatDate`. Does this change make sense to you?

Reply:
Sure, why not :) These functions were hard-coded to string before, but it would be nicer if they actually returned a date.